src/parser/createParser.ts

```diff
@@ -8,6 +8,8 @@ import grammar from './grammar.js';
 import LexerAdapter from './LexerAdapter.js';
 import { createEofToken } from '../lexer/token.js';
 
+const { Parser: NearleyParser, Grammar } = nearley;
```

Comment:
This is the most important change.

Now that our code is a full ESM module, NodeJS complains when attempting to import named things from CommonJS module. The workaround is to get the default export and destructure it manually.